src/sfdc46/salesforceBulk/csv/csv_delete_operator.bal

```diff
@@ -123,23 +123,30 @@ public type CsvDeleteOperator client object {
     # + waitTime - time between two tries in ms
     # + return - Batch result as CSV if successful else SalesforceError occured
     public remote function getBatchResults(string batchId, int numberOfTries = 1, int waitTime = 3000) 
-        returns @tainted string | SalesforceError {
+        returns @tainted Result[]|SalesforceError {
         int counter = 0;
         while (counter < numberOfTries) {
             Batch|SalesforceError batch = self->getBatchInfo(batchId);
             
             if (batch is Batch) {
+
                 if (batch.state == COMPLETED) {
-                    return self.httpBaseClient->getCsvRecord([JOB, self.job.id, BATCH, batchId, RESULT]);
+                    string|SalesforceError result = 
```

Comment:
Can we refactor this repetitive code segment to a separate function and call?

Reply:
Will fix in https://github.com/wso2-ballerina/module-salesforce/issues/63